llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp

```diff
@@ -2152,10 +2153,24 @@ void DwarfDebug::beginFunctionImpl(const MachineFunction *MF) {
   if (SP->getUnit()->getEmissionKind() == DICompileUnit::NoDebug)
     return;
 
-  DwarfCompileUnit &CU = getOrCreateDwarfCompileUnit(SP->getUnit());
-
-  Asm->OutStreamer->getContext().setDwarfCompileUnitID(
-      getDwarfCompileUnitIDForLineTable(CU));
+  if(SP->getUnit()->getEmissionKind() == llvm::DICompileUnit::CasFriendly) {
```

Comment:
Can you add a comment here that explains that this this what CasFriendly mode is doing here? For example, that we're creating a new CU per function, so each function gets its own linetable header?

Reply:
Of course!